tactics/ind_tables.ml

```diff
@@ -163,15 +164,17 @@ let define_mutual_scheme_base kind suff f mode names mind =
   let ids = Array.init (Array.length mib.mind_packets) (fun i ->
       try Int.List.assoc i names
       with Not_found -> add_suffix mib.mind_packets.(i).mind_typename suff) in
-  let consts = Array.map2 (fun id cl -> 
-     define mode id cl (Declareops.inductive_is_polymorphic mib) ctx) ids cl in
+  let fold (i, consts, effs) id cl =
+    let cst = define mode id cl (Declareops.inductive_is_polymorphic mib) ctx in
+    let role = Entries.Schema ((mind, i), kind)in
+    let neff = Safe_typing.private_constant (Global.safe_env ()) role cst in
+    (i + 1, cst :: consts, Safe_typing.concat_private neff effs)
+  in
+  let (_, consts, eff) = Array.fold_left2 fold (0, [], eff) ids cl in
```

Comment:
Why not use fold_left2_i instead of doing manual int manips?

Reply:
or even fold_left2_map_i since we're producing this `consts` array

Reply:
We're producing both `eff` and `const`, so this rules out `fold_left2_map_i`. As for the index, yes, I could use the `fold_left2_i` function I guess, but this is a detail.

Reply:
Also this is effectful, I think folding ensures the correct order, while `fold_map` is unclear.

Reply:
I feel like the _i function would make it clearer that we do nothing interesting with the index, at first glance I wondered if there might be some skipping (like when we fold over a context and skip the increment for letins).

>Also this is effectful, I think folding ensures the correct order, while fold_map is unclear.

Isn't the order specified since it says `left` in the function name?

Reply:
Right, if you *really* insist I can change that...